src/pages/HomePage.tsx

```diff
@@ -0,0 +1,151 @@
+import React, {useEffect, useMemo, useState} from 'react';
+import {Button, StyleSheet, Text, View} from 'react-native';
+import {Address, Chain, createWalletClient, custom, formatEther} from 'viem';
+import {mainnet, sepolia} from 'viem/chains';
+import {useWalletConnectModal} from '@walletconnect/modal-react-native';
+import {publicClient} from '../clients/public';
+
+export const CHAINS = [mainnet, sepolia];
+
+type ConnectEventInfo = {
```

Comment:
Do we have to declare our types here? Could we use WC ones?

Reply:
They do not export the types correctly here, but I was able to simplify and workaround it by doing:

```tsx
    const onConnectEvent = async ({ session }: { session: IProvider['session'] }) => { ... }
```